### PR TITLE
12.0 website product filters

### PR DESCRIPTION
**Pull request by krunal267**
Task URLs:

https://www.druidoo.io/web#id=1226&action=367&model=project.task&view_type=form&menu_id=237

https://www.druidoo.io/web#id=1227&action=367&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

**Comment by ivantodorovich** [comment]
Hi @krunal267 

Can you please fix travis errors?

```
website_product_attribute_filter/__manifest__.py:4:1: W191 indentation contains tabs
website_product_attribute_filter/__manifest__.py:4:1: E101 indentation contains mixed spaces and tabs
website_product_attribute_filter/__manifest__.py:4:80: E501 line too long (88 > 79 characters)
website_product_attribute_filter/__manifest__.py:5:1: W191 indentation contains tabs
website_product_attribute_filter/__manifest__.py:5:80: E501 line too long (102 > 79 characters)
website_product_attribute_filter/__manifest__.py:6:1: W191 indentation contains tabs
website_product_attribute_filter/__manifest__.py:6:77: W291 trailing whitespace
website_product_attribute_filter/__manifest__.py:7:1: W191 indentation contains tabs
website_product_attribute_filter/__manifest__.py:7:80: E501 line too long (99 > 79 characters)
website_product_attribute_filter/__manifest__.py:8:1: W191 indentation contains tabs
website_product_attribute_filter/__manifest__.py:9:1: E101 indentation contains mixed spaces and tabs
website_product_attribute_filter/controllers/main.py:3:1: F401 'odoo.SUPERUSER_ID' imported but unused
website_product_attribute_filter/controllers/main.py:5:1: F401 'odoo.addons.http_routing.models.ir_http.slug' imported but unused
website_product_attribute_filter/controllers/main.py:7:1: F401 'odoo.addons.website_sale.controllers.main.TableCompute' imported but unused
website_product_attribute_filter/controllers/main.py:8:1: F401 'odoo.addons.website_sale.controllers.main.QueryURL' imported but unused
website_product_attribute_filter/controllers/main.py:27:80: E501 line too long (109 > 79 characters)
website_product_attribute_filter/controllers/main.py:56:80: E501 line too long (92 > 79 characters)
website_product_attribute_filter/controllers/main.py:66:80: E501 line too long (198 > 79 characters)
website_sale_price_filter/controllers/main.py:25:80: E501 line too long (82 > 79 characters)
website_sale_price_filter/controllers/main.py:26:9: F841 local variable 'attributes_ids' is assigned to but never used
website_sale_price_filter/controllers/main.py:27:9: F841 local variable 'attrib_set' is assigned to but never used
website_sale_price_filter/controllers/main.py:41:80: E501 line too long (82 > 79 characters)
```




and:

```
************* Module website_sale_price_filter
website_sale_price_filter/__init__.py:1: [C7902(missing-readme), ] Missing ./README.rst file. Template here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
website_sale_price_filter/views/assets.xml:1: [E7902(xml-syntax-error), ]  error: Document is empty, line 1, column 1 (assets.xml, line 1)
website_sale_price_filter/views/assets.xml:1: [W7930(file-not-used), ]  not used from manifest
************* Module website_sale_price_filter.__manifest__
website_sale_price_filter/__manifest__.py:1: [C8103(manifest-deprecated-key), ] Deprecated key "description" in manifest file
************* Module website_product_attribute_filter
website_product_attribute_filter/__init__.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
website_product_attribute_filter/__init__.py:1: [C7902(missing-readme), ] Missing ./README.rst file. Template here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
website_product_attribute_filter/views/templates.xml:3: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:4: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:5: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:6: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:7: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:8: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:178: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/views/templates.xml:186: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/static/src/js/attribute.js:3: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
website_product_attribute_filter/static/src/js/attribute.js:8: [W7910(wrong-tabs-instead-of-spaces), ]  Use wrong tabs indentation instead of four spaces
************* Module website_product_attribute_filter.__manifest__
website_product_attribute_filter/__manifest__.py:1: [C8103(manifest-deprecated-key), ] Deprecated key "description" in manifest file
************* Module website_product_attribute_filter.controllers.main
website_product_attribute_filter/controllers/main.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
```

**Comment by ivantodorovich** [comment]
Also for price_filter, its style is not looking great:

image


does it look good in your local?

**Comment by ivantodorovich** [comment]
Also, when applying filters, sometimes the groups are empty:

image

This is because the attributes are no longer available in the current search result

I think this was solved in the view of the theme_ecox module (https://github.com/druidoo/ecox/blob/master/theme_ecox/views/website_sale_shop_filters.xml)

**Comment by ivantodorovich** [comment]
Hi @krunal267 

It looks like the css colors are hardcoded.
For instance, the color #a3498b is not using the .scss variable (`theme-color()`)

As a side-effect, changing the theme colors doesn't have an impact on the filters

image


To do so we need to change the `css` file to `scss`, and use the proper variables / functions.
If I'm not mistaken the function name is `theme-color( something )` where something is either `primary` or `secondary` (i don't remember that part)

Can you please check and make sure that all colors are using the proper scss variables?

Also there are a lot of styles defined on the views, like for instance this one:
https://github.com/druidoo/druidoo-addons/pull/24/files#diff-a0306b8e68da22128e8eebc1618a9429R3

Can you please move the styles (or at least the most relevant ones) to the style sheets?
And also make sure we're using scss variables whenever its needed.

Basically, make sure it works on different color styles / themes. It should be theme-compatible that way
